pkg/staticnetworkconfig/backward_compatibility.go

```diff
@@ -25,6 +25,61 @@ func (s *StaticNetworkConfigGenerator) NMStatectlServiceSupported(version string
 	return versionOK, nil
 }
 
+// CheckConfigForAutoDnsSetToFalse TODO: This is a temporary workaround and should be removed once the auto-dns:false bug is fixed
+func (s *StaticNetworkConfigGenerator) CheckConfigForAutoDnsSetToFalse(staticNetworkConfigStr string) (bool, error) {
+	staticNetworkConfig, err := s.decodeStaticNetworkConfig(staticNetworkConfigStr)
+	if err != nil {
+		s.log.WithError(err).Errorf("Failed to decode static network config")
+		return false, err
+	}
+
+	for _, hostConfig := range staticNetworkConfig {
+		isIncludeAutoDnsSetToFalse, err := s.hasAutoDnsSetToFalse(hostConfig.NetworkYaml)
+		if err != nil {
+			return false, err
+		}
+		if isIncludeAutoDnsSetToFalse {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (s *StaticNetworkConfigGenerator) hasAutoDnsSetToFalse(networksYaml string) (bool, error) {
+	var config map[string]interface{}
+
+	// Unmarshal the YAML string into the config struct
+	err := yaml.Unmarshal([]byte(networksYaml), &config)
+	if err != nil {
+		s.log.WithError(err).Errorf("Error unmarshalling yaml")
+		return false, err
+	}
+
+	interfaces, exists := config["interfaces"]
+	if !exists || interfaces == nil {
+		return false, nil
+	}
+	interfacesSlice, ok := interfaces.([]interface{})
+	if !ok {
+		return false, nil
+	}
+
+	for _, iface := range interfacesSlice {
+		nic := iface.(map[interface{}]interface{})
+		if val, exists := nic["ipv4"].(map[interface{}]interface{}); exists {
```

Comment:
just a suggestion. matter of style, but maybe could be a bit simplified with something like:
```
		isAutoDNSDisabled := func(ipConfig map[interface{}]interface{}) bool {
			if autoDns, exists := ipConfig["auto-dns"]; exists && autoDns == false {
				return true
			}
			return false
		}

		if ipv4, exists := nic["ipv4"].(map[interface{}]interface{}); exists && isAutoDNSDisabled(ipv4) {
			return true, nil
		}
		if ipv6, exists := nic["ipv6"].(map[interface{}]interface{}); exists && isAutoDNSDisabled(ipv6) {
			return true, nil
		}
```